ECLand: Initial CI test pipeline for GNU, Intel and MacOS in double-precision

mlange05 opened the pull request:
An initial test setup that runs CI in double-precision for some of the standard compilers. The install scripts are taken from ecRad and ecWAM, largely and simply execute the to standard tests via ctest.

Currently supported (and not yet supported):
* GNU-10 works fine with Ubuntu-provided Netcdf
* GNU-14 worked fine with Ubuntu-provided netcdf
* Intel-classic works fine with bootstrapped netcdf
* Nvidia compiles and run with bootstrapped netcdf, but ultimately fails tests - so disabled for now.
* MacOS/clang works fine with brew-installed netcdf

Please note that single precision failed to compile with several toolchains due to a utility routine (MINMAX) being called from precision-variant and precision-invariant code paths (with JPRB and JPRD). I've left the hooks in the workflow file for fixing in a separate PR.

awnawab commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,246 @@
+        include:
+
+          - name: linux gnu-10
+            os: ubuntu-20.04
```
We probably want to bump this up to ubuntu-22.04; 20.04 will be deprecated soon.

awnawab commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,246 @@
+            compiler_cc: gcc-10
+            compiler_cxx: g++-10
+            compiler_fc: gfortran-10
+            ctest_options: -V -E memory
```
Do we need these? I couldn't find a test called "memory", so not sure if we need this exclude here.

awnawab commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,246 @@
+          #   caching: true
+
+          - name : linux intel-classic
+            os: ubuntu-20.04
```
Same comment as above about ubuntu version.

awnawab commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,246 @@
+          sudo apt-get install libnetcdff-dev
+        fi
+
+        pip3 install numpy
```
Why do we need numpy?

awnawab reviewed (changes requested):
This looks great 👌 Thanks a lot for adding the CI setup and working through the fiddly netcdf issues 🙏  I've left just a few small comments that would be nice to cleanup before we merge.

reuterbal reviewed (changes requested):
This looks very good to me!

Only thing is, that I don't see any use of Python in the action, so you could skip the Python set-up step. Make sure to then also remove the python_version property from the matrix entries.

mlange05 commented:
Hi @reuterbal @awnawab @gabardu , sorry for taking so long to fix this. I've now incorporated all previous suggestions and fixed the build-type, so that for DP we have GNU-10, GNU-14 and Intel passing tests now. I've also reduced the verbosity of the long-integration test for debug purposes, as discussed offline. Please all, have another look so we can get these test in as a base reference for further updates.

@gabardu I didn't have to resort to per-compiler references in the end, as forcing the correct build type finally got things back on track. 😅 The Nvidia toolchain is still problematic, but I'll look into this separately.

reuterbal reviewed (approved):
Nice work, looks good to me!

awnawab reviewed (approved):
Thanks a lot for getting this over the line, looks great 👌

gabardu reviewed (approved):
Looks good to me!
